application/src/main/java/org/togetherjava/tjbot/commands/TopHelpers.java

```diff
@@ -0,0 +1,114 @@
+
+public final class TopHelpers implements Command {
+    private static final Logger logger = LoggerFactory.getLogger(TopHelpers.class);
+
+    private static final String COUNT_OPTION = "count";
+    private static final String NO_ENTRIES = "No entries";
+
+    private static final long MIN_COUNT = 10L;
+    private static final long MAX_COUNT = 30L;
+
+    private final Database database;
+
+    /**
+     * Initializes TopHelpers with a database.
+     *
+     * @param database the database to store the key-value pairs in
+     */
+    public TopHelpers(Database database) {
+        this.database = database;
+    }
+
+    @Override
+    public @NotNull String getCommandName() {
+        return "tophelpers";
+    }
+
+    @Override
+    public @NotNull String getDescription() {
+        return "Find top helpers for last 30 days";
+    }
+
+    @Override
+    public boolean isGuildOnly() {
+        return true;
+    }
+
+    @Override
+    public @NotNull CommandData addOptions(@NotNull CommandData commandData) {
+        return commandData.addOptions(new OptionData(OptionType.STRING, COUNT_OPTION,
+                "Count of top helpers to be retrieved (default is 10 and capped at 30)", false));
+    }
+
+    @Override
+    public void onSlashCommand(SlashCommandEvent event) {
+        var guildId = event.getGuild().getIdLong();
+        var count = getBoundedCount(event.getOption(COUNT_OPTION));
+        database.read(dsl -> {
+            var records = dsl.with("TOPHELPERS")
+                .as(select(MESSAGE_METADATA.USER_ID, count().as("COUNT")).from(MESSAGE_METADATA)
+                    .where(MESSAGE_METADATA.GUILD_ID.eq(guildId))
+                    .groupBy(MESSAGE_METADATA.USER_ID)
+                    .orderBy(count().desc())
+                    .limit(count))
+                .select(rowNumber().over(orderBy(field(name("COUNT")).desc())).as("#"),
+                        field(name("USER_ID"), Long.class), field(name("COUNT"), Long.class))
+                .from(table(name("TOPHELPERS")))
+                .fetch();
+            event.reply(prettyPrintOutput(buildTopHelpersDataFrame(event, records))).queue();
+        });
+    }
+
+    private long getBoundedCount(OptionMapping count) {
+        return count == null ? MIN_COUNT : Math.min(count.getAsLong(), MAX_COUNT);
+    }
+
+    private List<List<String>> buildTopHelpersDataFrame(SlashCommandEvent event,
+            Result<Record3<Integer, Long, Long>> records) {
+        return Arrays.stream(records.intoArrays())
+            .map(objects -> List.of(objects[0].toString(),
+                    getMemberNameFromEvent(event, objects[1]), objects[2].toString()))
+            .toList();
+    }
+
+    private String getMemberNameFromEvent(SlashCommandEvent event, Object member) {
+        String memberName;
+        try {
+            memberName = event.getGuild().getMemberById((Long) member).getEffectiveName();
```

Comment:
Doesn't work, there's a reason why I told to run the command instead of using tests.
Regarding why? Running it yourself and the docs explain why.

Reply:
Worked on my test server 🤷‍♂️ but I will check the doc